sql/src/main/java/io/crate/operation/collect/collectors/NodeStatsCollector.java

```diff
@@ -72,10 +73,24 @@ public NodeStatsCollector(TransportNodeStatsAction transportStatTablesAction,
     public void doCollect() {
         remainingRequests.set(nodes.size());
         final List<NodeStatsContext> rows = Collections.synchronizedList(new ArrayList<NodeStatsContext>());
+        Set<ColumnIdent> toCollect = topLevelColumnIdentVisitor.process(collectPhase.toCollect());
+        // check if toCollect only contains id and name, then it's not necessary to perform a request
+        boolean emitDirectly = true;
+        for (ColumnIdent ident : toCollect) {
```

Comment:
I think `toCollect.contains(ID) && toCollect.contains(NAME) && toCollect.size() == 2` would be more efficient.